src/main/java/com/mongodb/hibernate/internal/translate/AbstractMqlTranslator.java

```diff
@@ -301,7 +320,104 @@ public void visitParameter(JdbcParameter jdbcParameter) {
+        }
+
+        var collection = acceptAndYield(querySpec.getFromClause(), COLLECTION_NAME);
+
+        var whereClauseRestrictions = querySpec.getWhereClauseRestrictions();
+        var filter = whereClauseRestrictions == null || whereClauseRestrictions.isEmpty()
+                ? AstMatchesEverythingFilter.INSTANCE
+                : acceptAndYield(whereClauseRestrictions, FILTER);
+
+        var projectStageSpecifications = acceptAndYield(querySpec.getSelectClause(), PROJECT_STAGE_SPECIFICATIONS);
+
+        var stages = List.<AstStage>of(new AstMatchStage(filter), new AstProjectStage(projectStageSpecifications));
+        astVisitorValueHolder.yield(COLLECTION_AGGREGATE, new AstAggregateCommand(collection, new AstPipeline(stages)));
+    }
+
+    @Override
+    public void visitFromClause(FromClause fromClause) {
+        if (fromClause.getRoots().size() != 1) {
+            throw new FeatureNotSupportedException();
+        }
+        var tableGroup = fromClause.getRoots().get(0);
+
+        if (!(tableGroup.getModelPart() instanceof EntityPersister entityPersister)
+                || entityPersister.getQuerySpaces().length != 1) {
+            throw new FeatureNotSupportedException();
+        }
+
+        affectedTableNames.add(((String[]) entityPersister.getQuerySpaces())[0]);
+        tableGroup.getPrimaryTableReference().accept(this);
+    }
+
+    @Override
+    public void visitNamedTableReference(NamedTableReference namedTableReference) {
+        astVisitorValueHolder.yield(COLLECTION_NAME, namedTableReference.getTableExpression());
+    }
+
+    @Override
+    public void visitRelationalPredicate(ComparisonPredicate comparisonPredicate) {
+        var astComparisonFilterOperator = getAstComparisonFilterOperator(comparisonPredicate.getOperator());
+
+        var fieldPath = acceptAndYield(comparisonPredicate.getLeftHandExpression(), FIELD_PATH);
+        var fieldValue = acceptAndYield(comparisonPredicate.getRightHandExpression(), FIELD_VALUE);
+
+        var filter = new AstFieldOperationFilter(
+                new AstFilterFieldPath(fieldPath),
+                new AstComparisonFilterOperation(astComparisonFilterOperator, fieldValue));
+        astVisitorValueHolder.yield(FILTER, filter);
+    }
+
+    private static AstComparisonFilterOperator getAstComparisonFilterOperator(ComparisonOperator operator) {
+        return switch (operator) {
+            case EQUAL -> EQ;
+            default -> throw new FeatureNotSupportedException("Unsupported operator: " + operator.name());
+        };
+    }
+
+    @Override
+    public void visitSelectClause(SelectClause selectClause) {
+        var projectStageSpecifications = new ArrayList<AstProjectStageSpecification>(
+                selectClause.getSqlSelections().size());
+
+        for (SqlSelection sqlSelection : selectClause.getSqlSelections()) {
+            if (sqlSelection.isVirtual()) {
+                continue;
```

Comment:
There is some edge case that some `SqlSelection` entry exists for some reason but are not meant to be used in SQL. For such virtual entries, we simply ignore during SQL translation (a pattern also in `AbstractSqlTranslator`).